exporter/containerimage/annotations.go

```diff
@@ -0,0 +1,157 @@
+package containerimage
+
+import (
+	"fmt"
+
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/containerd/containerd/platforms"
+	"github.com/moby/buildkit/exporter/containerimage/exptypes"
+)
+
+type Annotations struct {
+	Index              map[string]string
+	IndexDescriptor    map[string]string
+	Manifest           map[string]string
+	ManifestDescriptor map[string]string
+}
+
+type AnnotationsGroup map[string]*Annotations
```

Comment:
This needs a comment like `// AnnotationsGroup is a map of annotations by the reference key`. Otherwise quite confusing.